ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -197,6 +198,9 @@ static void gatherStats(CompactionInfo ci, HiveConf conf, String userName, Strin
           statusUpdaterConf.set(TezConfiguration.TEZ_QUEUE_NAME, compactionQueueName);
         }
         SessionState sessionState = DriverUtils.setUpSessionState(statusUpdaterConf, userName, true);
+        Map<String, String> hiveVariables = sessionState.getHiveVariables();
+        hiveVariables.put(Constants.INSIDE_COMPACTION_TRANSACTION_FLAG, "true");
+        sessionState.setHiveVariables(hiveVariables);
```

Comment:
could we create sessionState.setHiveVariable() method in SessionState?

Reply:
Sure, I'll create it.